### PR TITLE
PP-5273 Write to Queue after DB transaction

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT YOU DID
- Updating/Committing charge status (for CAPTURE_APPROVED) before sending the charge to queue so that
    - queue message receiver won't receive the charge before the charge status commited to CAPTURE_APPROVED state.

Receiving charge from queue and processing (before charge being committed with CAPTURE_APPROVED state) will lead to below exception as charge will be read/updated concurrently by `CardCaptureService` and `CardCaptureMessageProcess`

`org.eclipse.persistence.exceptions.OptimisticLockException
The object [uk.gov.pay.connector.charge.model.domain.ChargeEntity@xxxxxx] cannot be updated because it has changed or been deleted since it was last read.`

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/paymentprocessor/service/CardCaptureService.java [hunk 1]
```diff
@@ -85,21 +81,13 @@ public ChargeEntity prepareChargeForCapture(String chargeId) {
...
-    @Transactional
     public ChargeEntity markChargeAsEligibleForCapture(String externalId) {
-        final ChargeEntity charge = chargeService.findChargeById(externalId);
+        ChargeEntity charge = chargeService.markChargeAsEligibleForCapture(externalId);
 
-        if (charge.isDelayedCapture()) {
-            return chargeService.updateChargeStatus(charge, AWAITING_CAPTURE_REQUEST);
-
-        } else {
+        if (!charge.isDelayedCapture())
```
Comment (jankowiakmaria) [inline src/main/java/uk/gov/pay/connector/paymentprocessor/service/CardCaptureService.java:87]: Would read better without exclamation mark, but functionally it's correct.
Would be better to have brackets around the if body, but can be addressed separately.

File: src/test/java/uk/gov/pay/connector/paymentprocessor/service/CardCaptureServiceTest.java [hunk 2]
```diff
@@ -494,12 +494,13 @@ public void markChargeAsEligibleForCapture_shouldThrowException_WithFeatureFlagE
                 mockConfiguration);
 
         String externalId = "external-id";
-        ChargeEntity charge = createNewChargeWith("worldpay", 1L, EXPIRED, "gatewayTxId");
+        ChargeEntity charge = createNewChargeWith("worldpay", 1L, AUTHORISATION_SUCCESS, "gatewayTxId");
```
Comment (jankowiakmaria) [inline src/test/java/uk/gov/pay/connector/paymentprocessor/service/CardCaptureServiceTest.java:497]: Not sure why this change is required.

**Review by jankowiakmaria — COMMENTED** [review]
One minor comment, one doubt comment. Other than that LGTM.

**Review by sfount — APPROVED** [review]

File: src/test/java/uk/gov/pay/connector/paymentprocessor/service/CardCaptureServiceTest.java (hunk 2 above)
Comment (kbottla) [reply src/test/java/uk/gov/pay/connector/paymentprocessor/service/CardCaptureServiceTest.java:497]: This was not right before. Expected is WebApplicationException but thrown due to charge being in invalid state `EXPIRED` but not when not able to add to queue

File: src/main/java/uk/gov/pay/connector/paymentprocessor/service/CardCaptureService.java (hunk 1 above)
Comment (kbottla) [reply src/main/java/uk/gov/pay/connector/paymentprocessor/service/CardCaptureService.java:87]: 👍 I will update it in the next

File: src/test/java/uk/gov/pay/connector/paymentprocessor/service/CardCaptureServiceTest.java (hunk 2 above)
Comment (jankowiakmaria) [reply src/test/java/uk/gov/pay/connector/paymentprocessor/service/CardCaptureServiceTest.java:497]: 👍 Thanks for explanation.
